boot/boot.go

```diff
@@ -158,6 +153,14 @@ func applicable(s snap.PlaceInfo, t snap.Type, dev snap.Device) bool {
 		if s.InstanceName() != base {
 			return false
 		}
+	case snap.TypeGadget:
+		// First condition: gadget is not a boot participant for UC16/18
+		// Second condition: a remodel might leave you in this state
```

Comment:
Can you expand a bit the comment for the second condition? Like, what is "this state" and why is it being checked here?

Reply:
I've clarified it a bit now

Reply:
It's still rather cryptic to me, but if you think that it's clear for someone who knows the domain, then I'm fine :-)

Reply:
I think "leave you" is fairly colloquial and I agree a bit unclear here. What is meant

"a remodel might leave behind installed a gadget that is not the device gadget anymore, ignore such a gadget checking the name"

same for base and kernel

Reply:
Changed

Reply:
Thanks!